### PR TITLE
feat: add basic `OnyxInput`

### PR DESCRIPTION
**Pull request by larsrickert**
relates to #189

Add a basic OnyxInput component.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    height: calc($line-height + 2 * $padding-vertical);
+    box-sizing: border-box;
+
+    &:has(.onyx-input__native:enabled:hover) {
+      border-color: var(--onyx-color-base-primary-400);
+    }
+
+    &:has(.onyx-input__native:enabled:focus) {
+      border-color: var(--border-color);
+      caret-color: var(--border-color);
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue:99]: I wouldn't change the caret-color, that could create a11y and annoy users.

Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue:99]: Thats the current design: https://www.figma.com/file/YfEUBOHk4J4nYrk04geswG/onyx---Design-System-Figma-Library?type=design&node-id=1005-170709&mode=dev

Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue:101]: But would be fine for me to remove the caret color

Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue]: I talked with Jannick, he is fine with removing it

**Pull request merged**